### PR TITLE
update for fillpeaks bug

### PR DESCRIPTION
**Pull request by jsaintvanne**
Hi, 
it refers to https://github.com/computational-metabolomics/msPurity/issues/67
just a proposition if we need to change for fillpeaks bugs
With this, I transform `NA` values into `0` and I have to do it before data matrix become a dataframe with the filename addition !
And I also have to change with `which` function to find the good peaks during RT matching

I don't know if this thing is an obligation and is good but if we need it it is just here :)

@yguitton @Tomnl @lecorguille

### CODE REVIEW

File: R/purityA-frag4feature.R [hunk 1]
```diff
@@ -225,9 +233,9 @@ fsub2  <- function(pro, allpeaks, intense, ppm, fullp=FALSE, use_grped=FALSE){
   }
 
   if (fullp){
```
Comment (Tomnl) [inline R/purityA-frag4feature.R]: Perhaps we can prevent converting to 0. And just skip the NA values in the filter stage e.g.

```
  if (fullp){
    rtmin_col <- "rtmin_full"
    rtmax_col <- "rtmax_full"
  }else{
    rtmin_col <- "rtmin"
    rtmax_col <- "rtmax"
  }

  mtchRT <- allpeaks[prt>=allpeaks[,rtmin_col] &
                     prt<=allpeaks[,rtmax_col] &
                     !is.na(allpeaks[,rtmin_col]) &
                     !is.na(allpeaks[,rtmax_col]),]
```

Comment (Tomnl) [reply R/purityA-frag4feature.R]: I thought the NAs are given from XCMS for the "filled" peaks? So I was just trying to preserve that information. 

My understanding is that if the "filled" peak is not peak picked (e.g. from centWave) certain columns cannot be calculated as so are given an NA value.

**Comment by Tomnl** [comment]
@jsaintvanne  - This seems to be passing on a simple unit-test - but could you try on your datasets to see if you get the expected result

**Comment by jsaintvanne** [comment]
@Tomnl 
Just rerun my tests with these modificaitons and it pass and I have the same results! 
That looks good for me ;) 

But the principal question is : is it really correct to do this? :) I think yes but not really sure at 100%

**Comment by Tomnl** [comment]
@jsaintvanne 

I am OK with the code so far (it doesn't seem the effect the previous functionality at least) but if you wanted to add anything else - please do so

**Comment by jsaintvanne** [comment]
Ho non it looks good for me also!! Just hope that don't break anything !!

**Comment by jsaintvanne** [comment]
Yes no problem !! 
Is it possible to merge it on the new release after it came out ?

**Comment by jsaintvanne** [comment]
> Potentially - as it is a bug fix rather than a new feature

Ho yes! Have to learn it to apply a bug fix on metaMS package x)
So now I'm waiting for the RELEASE to be able to deploy msPurity!! :)

**Pull request merged**